app-cpp/app355.cpp

```diff
@@ -68,38 +69,234 @@ class OrbitalData
+            // #1 std::ref() to force thread to take a reference& to a value; rather than a copy or pointer
+            // #2 Pass 'this' as first parameter since OnCalculateOrbitalDataMulti is an instance method
+            threadVector.push_back(std::thread(&SatOrbit::OnCalculateOrbitalDataMulti, this, tleBegin, tleEnd, std::ref(mOrbitalVector)));
+        }
+        // wait for the threads to finish before returning
+        std::for_each(threadVector.begin(), threadVector.end(), [](std::thread& inThread)
+        {
+            inThread.join();
+        });
+    }
+    return std::move(std::get<1>(mOrbitalVector));
+}
+
+void SatOrbit::SortOrbitalVector(std::vector<OrbitalData>& ioOrbitalVector)
+{
+    // Multithreaded sort
+    // If thread count is 1, use single threaded sort
+    if (mNumThreads == 1)
+    {
+        OnSortOrbitalVector(ioOrbitalVector);
+    }
+    else
+    {
+        const std::size_t orbitVectorSize = ioOrbitalVector.size();
+
+        // calculate the number of TLEs per thread
+        const std::size_t orbitPerThread = orbitVectorSize / mNumThreads;
+        // Note: no need for mutex as threads do not contend for shared elements during sort
+        IteratorPairVector subListVector{};
+
+        {
+            // create a list of threads the size of the number of threads specified
+            std::vector<std::thread> threadVector{};
+
+            for (std::size_t i = 0; i < mNumThreads; ++i)
+            {
+                // calculate the begin and end of the TLEs for the current thread
+                const auto orbitBegin = ioOrbitalVector.begin() + (i * orbitPerThread);
+                // if this is the last thread, add the remaining TLEs, else add orbitPerThread
+                const auto orbitEnd = (i == mNumThreads - 1) ? ioOrbitalVector.end() : orbitBegin + orbitPerThread;
+                threadVector.push_back(std::thread(&SatOrbit::OnSortOrbitalVectorMulti, this, orbitBegin, orbitEnd));
+
+                IteratorPairVector::value_type slice{ orbitBegin, orbitEnd };
+                subListVector.push_back(slice);
+            }
+            
+            // Join the threads together
+            // wait for the threads to finish before returning
+            std::for_each(threadVector.begin(), threadVector.end(), [](std::thread& inThread)
+            {
+                inThread.join();
+            });
+        }
+
+        // Merge the sorted sublists
+        // Use multithreading by merging multilple sublists at a time
+        // Join the threads before merging the set of larger sublists
+        // Loop until there is only one list of completely sorted OrbitalData
+        while (subListVector.size() > 1)
+        {
+            // Merge sort the sublists 2 at a time
+            std::vector<std::thread> threadVector{};
+            IteratorPairVector newSubListVector{};
+
+            for (std::size_t i = 1; i < subListVector.size(); i += 2)
```

Comment:
This is tricky!
You are starting index an `1` and steeping thru the indices by `+= 2`...
Add a comment above this `for()` loop explaining the intent.